server/src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -696,6 +696,41 @@ static String getCommitCheckpointFileName(long generation) {
         return TRANSLOG_FILE_PREFIX + generation + CHECKPOINT_SUFFIX;
     }
 
+    /**
+     * Trims translog for terms of files below <code>belowTerm</code> and seq# above <code>aboveSeqNo</code>.
+     * Effectively it moves max visible seq# {@link Checkpoint#trimmedAboveSeqNo} therefore {@link TranslogSnapshot} skips those operations.
+     */
+    public void trimOperations(long belowTerm, long aboveSeqNo) throws IOException {
+        assert aboveSeqNo >= 0 : "aboveSeqNo has to be non-negative number";
+
+        try (ReleasableLock lock = writeLock.acquire()) {
+            ensureOpen();
+            if (current.getPrimaryTerm() < belowTerm) {
+                throw new IllegalArgumentException("Trimming the translog can only be done for terms lower than the current one. " +
+                    "Trim requested for term [ " + belowTerm + " ] , current is [ " + current.getPrimaryTerm() + " ]");
+            }
+            // assume that there are no any trimmable ops in current
```

Comment:
nit: we assume that the current translog generation doesn't have trimmable ops. Verify that.